python/cudf/cudf/core/index.py

```diff
@@ -2811,57 +2813,80 @@ def interval_range(
             "Of the four parameters: start, end, periods, and "
             "freq, exactly three must be specified"
         )
-    if not isinstance(start or freq or end, int) and not isinstance(
-        start or freq or end, float
+    args = [start, end, freq, periods]
+    *args, periods = [cudf.Scalar(x) if x is not None else None for x in args]
+    if any(
+        [
+            not is_numerical_dtype(x.dtype) if x is not None else False
+            for x in args
+        ]
```

Comment:
I don't think we need the list here since `any(...)` can work with a generator expression